Model/DataObject/ClassDefinition/Data/ObjectBridge.php

```diff
@@ -240,9 +240,14 @@ public function getDataForEditmode($data, $object = null, $params = [])
                         }
                     }
 
-                    $bridgeFieldGetter = 'get' . $this->getBridgeField();
+                    $bridgeFieldName = $this->getBridgeField();
+                    $bridgeFieldGetter = 'get' . $bridgeFieldName;
                     $sourceObject = $bridgeObject->$bridgeFieldGetter();
 
+                    if(empty($sourceObject)){
+                        throw new \RuntimeException("Productsite $bridgeFieldName can not be empty for object with id ".$bridgeObject->getId());
```

Comment:
this is probably more accurate without 'Productsite'

Reply:
I agree